LeaderboardBackend.Test/Leaderboards.cs

```diff
@@ -35,24 +35,15 @@ public async Task OneTimeSetUp()
     }
 
     [OneTimeTearDown]
-    public void OneTimeTearDown()
-    {
-        _factory.Dispose();
-    }
+    public void OneTimeTearDown() => _factory.Dispose();
 
     [Test]
-    public void GetLeaderboard_NotFound()
-    {
-        RequestFailureException e = Assert.ThrowsAsync<RequestFailureException>(
-            async () =>
-                await _apiClient.Get<LeaderboardViewModel>(
-                    $"/api/leaderboard/{long.MaxValue}",
-                    new()
-                )
-        )!;
-
-        Assert.AreEqual(HttpStatusCode.NotFound, e.Response.StatusCode);
-    }
+    public async Task GetLeaderboard_NotFound() => await _apiClient.Awaiting(a => a.Get<LeaderboardViewModel>(
+        $"/api/leaderboard/{long.MaxValue}",
+        new()
+    )).Should()
```

Comment:
I would suggest using an `async` lambda and awaiting the call.

Reply:
But the test works right now so why would we want to switch to a lambda? That being said; how would I do that? Like this?

```cs
public void GetLeaderboard_NotFound() {
    Func<Task> act = () => _apiClient.Awaiting(a => a.Get<LeaderboardViewModel>(
        $"/api/leaderboard/{long.MaxValue}",
        new()
    ))
    await act.Should().ThrowAsync<RequestFailureException>()
        .Where(e => e.Response.StatusCode == HttpStatusCode.NotFound);
}
```

Reply:
No I just mean await the call to Get using an async anonymous function. Just a suggestion.

Reply:
Sure, but jsyk it's not actually needed; it behaves correctly. Even the example in the docs doesn't explicitly await it:
```cs
// From https://fluentassertions.com/exceptions/
Func<Task> act = asyncObject.Awaiting(x => x.ThrowAsync<ArgumentException>());
await act.Should().ThrowAsync<ArgumentException>();
```